src/chainstate/burn/operations/peg_in.rs

```diff
@@ -76,11 +78,17 @@ impl PegInOp {
 
             0      2  3                  24                            64       80
             |------|--|------------------|-----------------------------|--------|
-             magic  op   Stacks address      Contract name (optional)     memo
+             magic  op   Stacks address    Size prefixed contract name    memo
+                                               (optional, see note)
+
+             # Notes
 
-             Note that `data` is missing the first 3 bytes -- the magic and op must
+             `data` is missing the first 3 bytes -- the magic and op must
              be stripped before this method is called. At the time of writing,
              this is done in `burnchains::bitcoin::blocks::BitcoinBlockParser::parse_data`.
+
+             Contract name is prefixed by a single byte size followed by the UTF-8 encoded
```

Comment:
Technically speaking we're only allowing a subset of ascii in the contract name. Although ascii is a subset of UTF-8, hence why we can use utf8 to decode the string.